app/controllers/V1/searches_controller.rb

```diff
@@ -0,0 +1,112 @@
+module V1
+  class SearchesController < ApplicationController
+    def create
+      data = search_query
+      raw_data = Submission.where(id: data.pluck(:id))
+
+      render json: build_results(data, raw_data), status: :created
+    rescue StandardError => e
+      render json: { message: "AppStore search query raised #{e.message}" }, status: :unprocessable_entity
+    end
+
+  private
+
+    def search_query
+      relation = Search.where(date_submitted: (submitted_from..submitted_to))
+      relation = relation.where(date_updated: (updated_from..updated_to))
+      relation = relation.where(submission_type:) if submission_type
+      relation = relation.where(status:) if status
+      relation = relation.where("has_been_assigned_to ? :caseworker_id", caseworker_id:) if caseworker_id
+      relation = relation.order(**sort_clause)
+
+      relation.where_terms(query)
+    end
+
+    def build_results(data, raw_data)
+      {
+        metadata: {
+          total_results: data.size,
+          page: offset,
+          per_page: limit,
+        },
+        data: data.limit(limit).offset(offset),
+        raw_data: raw_data.limit(limit).offset(offset),
+      }.to_json
+    end
+
+    def submission_type
+      search_params[:submission_type]
+    end
+
+    def submitted_from
+      search_params[:submitted_from]&.to_date
+    end
+
+    def submitted_to
+      search_params[:submitted_to]&.to_date
+    end
+
+    def updated_from
+      search_params[:updated_from]&.to_date
+    end
+
+    def updated_to
+      search_params[:updated_to]&.to_date
+    end
+
+    def status
+      search_params[:status]
+    end
+
+    def caseworker_id
+      search_params[:caseworker_id]
+    end
+
+    def query
+      search_params[:query]
+    end
+
+    def sort_clause
+      return { date_updated: :desc } unless search_params[:sort_by]
+
+      { sort_by => sort_direction }
+    end
+
+    def sort_by
+      search_params.fetch(:sort_by, :date_updated)
+    end
+
+    def sort_direction
+      @sort_direction ||= search_params
+                            .fetch(:sort_direction, "asc")
+                            .downcase
+                            .gsub("ascending", "asc")
+                            .gsub("descending", "desc")
+    end
+
+    def limit
+      search_params.fetch(:per_page, 10).to_i
+    end
+
+    def offset
+      search_params.fetch(:page, 0).to_i
+    end
+
+    def search_params
+      params.permit(
+        :query,
+        :page,
+        :per_page,
+        :sort_by,
+        :sort_direction,
+        :submission_type,
```

Comment:
there is a fair bit of naming issue relating to submission versus application and application versus claim. In the context of app store that is even more confused due to the underlying table names and model names, not to mention controller names (ApplicationController). I prefer submission type to avoid the ambiguity, at least in the app store, but ideally in the caseworker too.

Reply:
So the reason I went with `application_type` was that that's what the field is in the app store database, the caseworker database and in all the payloads that currently go back and forth between app store and provider/caseworker. Calling it `submission_type` in this one instance feels odd, even though I agree it's a better name overall. But I don't want to bikeshed so I'll change on my end.

Reply:
its ok, i've changed my end now